src/main/scala/diplomaticobjectmodel/logicaltree/LogicalTrees.scala

```diff
@@ -32,45 +33,19 @@ class CLINTLogicalTreeNode(device: SimpleDevice, f: => OMRegisterMap) extends Lo
   }
 }
 
-class DebugCSRsLogicalTreeNode(
-  device: SimpleDevice,
-  f: => OMRegisterMap,
-  p: Parameters
-) extends LogicalTreeNode {
-  def getOMDebug(resourceBindings: ResourceBindings): Seq[OMComponent] = {
-    val memRegions: Seq[OMMemoryRegion] = DiplomaticObjectModelAddressing
-      .getOMMemoryRegions("Debug", resourceBindings, Some(f))
-    val cfg: DebugModuleParams = p(DebugModuleParams)
-
-    Seq[OMComponent](
-      OMDebugCSRs(
-        nDebugScratchRegisters = 0,
-        nTriggers = 0,
-        supportedOMTriggerTypes = Nil, // List[List[OMTriggerType]], // Each trigger could support different types
-        nTriggerChainDepth = 0,
-        dscrXdebugver = 0,
-        modeMTdataAccessible = Nil, // List[Boolean],
-        hasMcontrolHit = Nil, //  List[Boolean],
-        supportedMcontrolActions = Nil //  List[TriggerMatchAction], // What to do if trigger hits
-      )
-    )
-  }
-
-  def getOMComponents(resourceBindingsMap: ResourceBindingsMap, components: Seq[OMComponent]): Seq[OMComponent] = {
-    DiplomaticObjectModelAddressing.getOMComponentHelper(device, resourceBindingsMap, getOMDebug)
-  }
-}
-
 class DebugLogicalTreeNode(
   device: SimpleDevice,
-  f: => OMRegisterMap,
-  p: Parameters,
-  m: () => TLDebugModule
-) extends LogicalTreeNode {
+  dmOuter: () => TLDebugModuleOuterAsync,
+  dmInner: () => TLDebugModuleInnerAsync
+)(implicit val p: Parameters) extends LogicalTreeNode {
   def getOMDebug(resourceBindings: ResourceBindings): Seq[OMComponent] = {
+    val nComponents: Int = dmOuter().dmOuter.module.getNComponents()
```

Comment:
Above comment would mean you don't need to keep calling the nested versions here.

Reply:
This code is not executed until after the test harness memories are instantiated and control returns to the EnterpriseGenerator. @terpstra knows about this problem